src/python/zquantum/optimizers/utils.py

```diff
@@ -69,6 +69,8 @@ def save_optimization_results(optimization_results, filename):
                 data[key] = optimization_results[key].tolist()
             elif type(optimization_results[key]) == bytes:
                 data[key] = optimization_results[key].decode("utf-8")
+            elif type(optimization_results[key]).__module__ == np.__name__:
+                data[key] = optimization_results[key].item()
```

Comment:
Can you explain what is going on here?

Reply:
Numpy number types are not serializable – we need to do this in order to get `int` or `float` type out of `np.int32` or `np.float64` data types.